dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -0,0 +1,75 @@
+class Api::V1::UserSchoolInfosController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user_school_info
+
+  skip_before_action :verify_authenticity_token
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    if @user_school_info.user.id == current_user.id
+      if @user_school_info.update(last_confirmation_date: DateTime.now)
```

Comment:
I don't think this does what you expect it to. [`ActiveRecord.update` returns the resulting object whether it was successfully updated or not](https://apidock.com/rails/ActiveRecord/Base/update/class)

Reply:
Good catch!  I switched to update! so an error will be thrown if the update is unsuccessful.